src/foam/nanos/client/ClientBuilder.js

```diff
@@ -56,26 +57,33 @@ foam.CLASS({
             package: 'foam.nanos.client',
             name: 'Client',
             exports: [],
-            properties: [],
+            properties: []
           };
 
           var references = [];
 
           // Force hard reload when app version updates
-          self.nSpecDAO.find("appConfig").then(function(spec) {
-            var appConfig = spec.service;
-
-            client.exports.push(spec.name);
+          var appConfigService = self.ClientAppConfigService.create({
+            delegate: self.RetryBox.create({
+              maxAttempts: 2,
+              delegate: self.HTTPBox.create({
+                method: 'POST',
+                url: 'service/appConfigService'
+              })
+            })
+          });
+          var appConfigPromise = appConfigService.getAppConfig().then(function(appConfig) {
```

Comment:
I don't think we need to build this service. Won't it get created along with the others from the nSpecDAO?

Reply:
Reworked it a little to use nSpecDAO